harness/determined/exec/harness.py

```diff
@@ -155,6 +155,7 @@ def _run_pytorch_trial(
 
             train_context._set_default_gradient_compression(fp16_compression)
             train_context._set_default_average_aggregated_gradients(average_aggregated_gradients)
+            train_context._set_is_pre_trainer()
```

Comment:
pre-trainer seems a bit weird as a name, harness is a separate codepath which will still exist for a long time post-trainer. `is_harness()`?